### PR TITLE
feat: start persisting teams rule settings

### PR DESCRIPTION
**Pull request by rexscaria**
This fixes gateway rule settings not acting correctly because they were never properly persisted in the state.

### CODE REVIEW

**Comment by jacobbednarz** [comment]
looks like we've hit a regression here

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsRule_" -count 1 -timeout 120m -parallel 1

=== RUN   TestAccCloudflareTeamsRule_Basic
--- PASS: TestAccCloudflareTeamsRule_Basic (2.70s)
=== RUN   TestAccCloudflareTeamsRule_NoSettings
    resource_cloudflare_teams_rules_test.go:98: Step 2/2 error: Check failed: Check 8/8 error: cloudflare_zero_trust_gateway_policy.anszqvautt: Attribute 'rule_settings.#' expected "0", got "1"
--- FAIL: TestAccCloudflareTeamsRule_NoSettings (3.20s)
=== RUN   TestAccCloudflareTeamsRule_CustomResolver
--- PASS: TestAccCloudflareTeamsRule_CustomResolver (2.46s)
=== RUN   TestAccCloudflareTeamsRule_WithClipboardRedirection
--- PASS: TestAccCloudflareTeamsRule_WithClipboardRedirection (2.36s)
FAIL
FAIL	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	12.218s
FAIL
make: *** [testacc] Error 1
```

**Comment by rexscaria** [comment]
Updated test. Apologies  Thanks

**Comment by jacobbednarz** [comment]
nice one

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareTeamsRule_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareTeamsRule_Basic
--- PASS: TestAccCloudflareTeamsRule_Basic (3.18s)
=== RUN   TestAccCloudflareTeamsRule_NoSettings
--- PASS: TestAccCloudflareTeamsRule_NoSettings (3.18s)
=== RUN   TestAccCloudflareTeamsRule_CustomResolver
--- PASS: TestAccCloudflareTeamsRule_CustomResolver (2.29s)
=== RUN   TestAccCloudflareTeamsRule_WithClipboardRedirection
--- PASS: TestAccCloudflareTeamsRule_WithClipboardRedirection (2.21s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	11.383s
```

File: internal/sdkv2provider/resource_cloudflare_teams_rules_test.go [hunk 1]
```diff
@@ -116,7 +116,10 @@ func TestAccCloudflareTeamsRule_NoSettings(t *testing.T) {
 					resource.TestCheckResourceAttr(name, "action", "block"),
 					resource.TestCheckResourceAttr(name, "filters.0", "dns"),
 					resource.TestCheckResourceAttr(name, "traffic", "any(dns.domains[*] == \"example.com\")"),
-					resource.TestCheckResourceAttr(name, "rule_settings.#", "0"),
+					resource.TestCheckResourceAttr(name, "rule_settings.#", "1"),
+					resource.TestCheckResourceAttr(name, "rule_settings.#", "1"),
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_teams_rules_test.go:120]:
```suggestion
```

File: .changelog/4261.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:dependency
```
Comment (jacobbednarz) [inline .changelog/4261.txt:1]:
~~~suggestion
```release-note:fix
~~~

**Pull request merged**